### PR TITLE
Added realtime move history panel with highlighting

### PR DESCRIPTION
**Pull request by kesril**
- [x] Display moves in a scrollable list next to the chessboard

- [x] Show move numbers (1. e4 e5, 2. Nf3...)

- [x] Update the display in real-time as moves are made

- [x] (Optional) Highlight current position in the move list

- [x] (Optional) Allow clicking on moves to jump to that position (WILL DO)

### CODE REVIEW

**Review by jorge1289 — DISMISSED** [review]
I noticed your PR adds new functionality to game.js. We currently have another PR in the queue that modified the same file with JSDocs changes. To avoid any potential merge conflicts, could you hold off on merging until that PR is complete? Once it's in, please rebase your branch on the updated main to incorporate those doc changes. This will help ensure both sets of changes integrate smoothly.

**Comment by kesril** [comment]
Added clicking on moves that jump to that position. Will need re-approval!

**Review by jorge1289 — APPROVED** [review]
Same comment as previous commit. Good work!

**Pull request merged**
